cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -715,6 +766,14 @@ protected KafkaStatus createStatus(Kafka kafka) {
             status.setClusterId(kafka.getStatus().getClusterId());
         }
 
+        // We copy the versions if set
+        if (kafka.getStatus() != null && kafka.getStatus().getOperatorLastSuccessfulVersion() != null)  {
+            status.setOperatorLastSuccessfulVersion(kafka.getStatus().getOperatorLastSuccessfulVersion());
+        }
+        if (kafka.getStatus() != null && kafka.getStatus().getKafkaVersion() != null)  {
+            status.setKafkaVersion(kafka.getStatus().getKafkaVersion());
+        }
```

Comment:
If you copy the versions first here ... why is the code on line 162-182 needed? It seems its one or the other.

Reply:
I copied the precedent set by clusterID field (slightly above my changes)
I assume it has been set like this to maintain the behaviour so that it works if someone calls `reconcile` or `createOrUpdate` as if you call `createOrUpdate` directly I belive it skips `createStatus`.

I am happy to keep it to just the one place to do the check, but if that's the case I believe `createOrUpdate` may be the best place, thoughts?

Reply:
I would be careful about touching the cluster ID because it might break the cluster. But I'm not sure why would it need to be done twice.